framework/src/wifi_manager/wifi_manager.c

```diff
@@ -398,25 +514,50 @@ wifi_manager_result_e wifi_manager_set_mode(wifi_manager_mode_e mode, wifi_manag
 		result = wifi_utils_start_sta();
 		if (result != WIFI_UTILS_SUCCESS) {
 			ndbg("start STA fail (change STA mode fail)");
-			wifi_mutex_release(&w_mutex);
+			wifi_mutex_release(w_mutex);
 			return WIFI_MANAGER_FAIL;
 		}
 
 		g_manager_info.mode = STA_MODE;
-		g_manager_info.status = AP_DISCONNECTED;
-		wifi_mutex_release(&w_mutex);
+		wifi_status_set(AP_DISCONNECTED);
+		wifi_mutex_release(w_mutex);
 
-		nvdbg("Wifi Chnage SOFT AP MODE -> STA MODE");
+		nvdbg("Wifi mode is changed to station");
+	}
+	else {
```

Comment:
I guess this could be shown in svace as unreached codes because of "if (mode != STA_MODE && mode != SOFTAP_MODE) { ... return WIFI_MANAGER_INVALID_ARGS } "
on top of this function